python/BioSimSpace/Sandpit/Exscientia/FreeEnergy/_restraint.py

```diff
@@ -416,19 +421,27 @@ def format_bond(equilibrium_values, force_constants):
             )
 
         # Format the parameters for the angles and dihedrals
-        def format_angle(equilibrium_values, force_constants):
+        def format_angle(equilibrium_values, force_constants, restraint_lambda):
             converted_equ_val = (
                 self._restraint_dict["equilibrium_values"][equilibrium_values] / _degree
             )
             converted_fc = self._restraint_dict["force_constants"][force_constants] / (
                 _kj_per_mol / (_radian * _radian)
             )
-            return parameters_string.format(
-                eq0="{:.3f}".format(converted_equ_val),
-                fc0="{:.2f}".format(0),
-                eq1="{:.3f}".format(converted_equ_val),
-                fc1="{:.2f}".format(converted_fc),
-            )
+            if restraint_lambda:
+                return dihedral_restraints_parameters_string.format(
```

Comment:
Unnecessary duplication, could have something like:
```
par_string = dihedral_restraints_parameters_string if restraint_lambda else parameters_string
string.format(...)
```

Reply:
Nah, we cannot do that because this is only for dihedral and we use the same parameters_string for bond and angle , where they don't need this modification.